### PR TITLE
treewide: cleanup Adobe Flash Player

### LINKED ISSUE #110314: Cleanup Adobe Flash Player plugin support in Nixpkgs
## Progress (copied from @taku0's comment below)

Related packages:

- [ ] `pkgs/applications/networking/browsers/mozilla-plugins/flashplayer/`
- [x] `pkgs/applications/networking/browsers/chromium/plugins.nix`
- [ ] `pkgs/os-specific/linux/hal-flash`
- [ ] `pkgs/applications/networking/browsers/firefox/wrapper.nix`
- [x] `pkgs/applications/networking/browsers/chromium/default.nix`
- [ ] `pkgs/development/libraries/qt-5/modules/qtwebkit.nix` (`flashplayerFix`)
- [ ] `pkgs/development/libraries/qt-4.x/4.8/default.nix` (`flashplayerFix`)

Related documents:

- [ ] `doc/using/configuration.xml`
- [ ] `nixos/doc/manual/configuration/customizing-packages.xml`

## Chromium

I haven't looked into this but the Flash support should be completely removed since Chromium 88+ (#110010):
> Flash support/ capability will be complete removed from Chromium.  It will no longer be possible to enable Flash Player with Enterprise policy in Chrome 88+.
> **Source**: https://www.chromium.org/flash-roadmap#TOC-Flash-Support-Removed-from-Chromium-Target:-Chrome-88---Jan-2021-
> **Also**: https://chromestatus.com/feature/5643527180517376

So I guess we can cleanup `pkgs/applications/networking/browsers/chromium/plugins.nix` and the related logic (`enablePepperFlash`) now?

cc @taku0

### PR DESCRIPTION
**Pull request by lovesegfault**
###### Motivation for this change
Fixes: #110314

###### Things done
```
- firefox: remove flash support
- lib: remove mention of flashplayer in docs
- doc: remove mentions of flashplayer
- stdenv: remove mention of flashplayer (in comments)
- flashplayer: drop
- hal-flash: drop
- qt48: remove flashplayerFix
- qt5: remove flashplayerFix from qtwebkit
- nixos.flashpolicyd: drop
```

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/browsers/firefox/wrapper.nix [hunk 1]
```diff
@@ -47,28 +46,28 @@ let
...
+            "enableGoogleTalkPlugin"
             "enableMPlayer"
-            "jre"
+            "enableVLC"
             "icedtea"
-            "enableGoogleTalkPlugin"
-            "enableFriBIDPlugin"
-            "enableBluejeans"
-            "enableAdobeReader"
+            "jre"
```
Comment (lovesegfault) [inline pkgs/applications/networking/browsers/firefox/wrapper.nix:67]: I sorted this list when I added `enableAdobeFlash`, which is why this chunk of the diff is weird.

File: nixos/modules/module-list.nix [hunk 2]
```diff
@@ -650,7 +650,6 @@
   ./services/networking/fireqos.nix
   ./services/networking/firewall.nix
   ./services/networking/flannel.nix
-  ./services/networking/flashpolicyd.nix
```
Comment (samueldr) [inline nixos/modules/module-list.nix:653]: We'll need release notes changes for the removal of the module.

(And also for every other changes at once as a highlight of the release I guess.)

Comment (lovesegfault) [reply nixos/modules/module-list.nix:653]: Done!

File: nixos/doc/manual/release-notes/rl-2103.xml [hunk 3]
```diff
@@ -466,6 +466,21 @@ self: super:
...
    </listitem>
+   <listitem>
+    <para>
+     Adobe Flash Player support has been dropped from the tree. In particular,
+     the following packages no longer support it:
+     <itemizedlist>
+      <listitem><package>chromium</package></listitem>
+      <listitem><package>firefox</package></listitem>
+      <listitem><package>qt48</package></listitem>
+      <listitem><package>qt5.qtwebkit</package></listitem>
```
Comment (jtojnar) [inline nixos/doc/manual/release-notes/rl-2103.xml:477]: [List items](https://tdg.docbook.org/tdg/5.1/listitem.html) need to contain a paragraph of sorts.

```suggestion
      <listitem><simpara><package>chromium</package></simpara></listitem>
      <listitem><simpara><package>firefox</package></simpara></listitem>
      <listitem><simpara><package>qt48</package></simpara></listitem>
      <listitem><simpara><package>qt5.qtwebkit</package></simpara></listitem>
```

Comment (lovesegfault) [reply nixos/doc/manual/release-notes/rl-2103.xml]: Fixed!

File: pkgs/applications/networking/browsers/firefox/wrapper.nix (hunk 1 above)
Comment (ajs124) [reply pkgs/applications/networking/browsers/firefox/wrapper.nix:67]: Do any of these still work? Those are all NPAPI plugins, right? Those haven't been working [for a while](https://support.mozilla.org/en-US/kb/npapi-plugins).

Not saying this should be part of this PR, just wondering if we should drop all of this other stuff like `enableGoogleTalk`, `enableAdobeReader`, etc. as well and if you'd be interested in doing this as a follow-up PR.

Comment (ajs124) [reply pkgs/applications/networking/browsers/firefox/wrapper.nix:67]: Oops, nevermind… I didn't read enough context sorry for the noise.

**Comment by samueldr** [comment]
Some things that I didn't think about earlier.

 * Should the module get an `mkRemovedOptionModule`? 
 * Should the packages removed be aliased with a throw?

I'm not entirely sure whether we have a written policy about this. I personally think that any removed attribute should get aliased accordingly to the type of change. Same for modules.

Anyone else having opinions for/against?

**Comment by lovesegfault** [comment]
> Some things that I didn't think about earlier.
> 
>     * Should the module get an `mkRemovedOptionModule`?
> 
>     * Should the packages removed be aliased with a throw?
> 
> 
> I'm not entirely sure whether we have a written policy about this. I personally think that any removed attribute should get aliased accordingly to the type of change. Same for modules.
> 
> Anyone else having opinions for/against?

Done, aliases and `mkRemovedOptionModule` added

**Review by jtojnar — APPROVED** [review]

**Review by samueldr — APPROVED** [review]

**Review by taku0 — APPROVED** [review]
Code-wise LGTM.

**Pull request merged**